### PR TITLE
Made output of error more specific to bad parameter

### PR DESCRIPTION
**Pull request by oatmealraisin**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1199391

Changed the returned field.NotFound error so that it returns a more specific description of the cause of the error.

Previous:
 error: error processing the template "ruby-helloworld-sample": Template "ruby-helloworld-sample" is invalid: template.parameters[0]: Not found: {"Name":"ADMIN_USERNAME","DisplayName":"","Description":"administrator
           : username","Value":"","Generate":"test","From":"admin[A-Z0-9]{3}","Required":false}

Current:
error: error processing the template "ruby-helloworld-sample": Template "ruby-helloworld-sample" is invalid: template.parameters[0].Generate: Not found: "test"

@bparees

### CODE REVIEW

File: pkg/template/template.go [hunk 1]
```diff
@@ -163,7 +163,7 @@ func (p *Processor) GenerateParameterValues(t *api.Template) *field.Error {
 		if param.Generate != "" {
 			generator, ok := p.Generators[param.Generate]
 			if !ok {
-				return field.NotFound(templatePath, param)
+				return field.NotFound(templatePath.Child("Generate"), param.Generate)
```
Comment (bparees) [inline pkg/template/template.go]: let's switch it to a field invalid error instead of a notfound error.

**Comment by oatmealraisin** [comment]
@bparees How is this?

File: pkg/template/template.go [hunk 2]
```diff
@@ -163,7 +163,7 @@ func (p *Processor) GenerateParameterValues(t *api.Template) *field.Error {
 		if param.Generate != "" {
 			generator, ok := p.Generators[param.Generate]
 			if !ok {
-				return field.NotFound(templatePath, param)
+				return field.Invalid(templatePath.Child("Generate"), param.Generate, "")
```
Comment (bparees) [inline pkg/template/template.go]: Invalid takes an additional argument (details about what is invalid about the value)

you can actually copy the message from line 169/170 pretty much.

**Comment by bparees** [comment]
something has gone very wrong w/ this PR (you've picked up a bunch of other commits)

**Pull request closed without merging**